docs/devel/coding-conventions.md

```diff
@@ -30,12 +30,48 @@ Documentation for other releases can be found at
 <!-- END STRIP_FOR_RELEASE -->
 
 <!-- END MUNGE: UNVERSIONED_WARNING -->
-Coding style advice for contributors
+Code conventions
   - Bash
     - https://google-styleguide.googlecode.com/svn/trunk/shell.xml
   - Go
-    - https://github.com/golang/go/wiki/CodeReviewComments
-    - https://gist.github.com/lavalamp/4bd23295a9f32706a48f
+    - Ensure your code passes the [presubmit checks](development.md#hooks)
+    - [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+    - [Effective Go](https://golang.org/doc/effective_go.html)
+    - Command-line flags should use dashes, not underscores
+    - API conventions
+      - [API changes](api_changes.md)
+      - [API conventions](api-conventions.md)
+    - [Kubectl conventions](kubectl-conventions.md)
+    - [Logging conventions](logging.md)
+
+Testing conventions
+  - All new packages and most new significant functionality must come with unit tests
+  - Table-driven tests are preferred for testing multiple scenarios/inputs; for example, see [TestNamespaceAuthorization](../../test/integration/auth_test.go)
```

Comment:
Unit tests must pass on Mac and Windows platforms - if you use linux specific features, your test case must either be skipped on windows or compiled out (skipped is better when running linux specific commands, compiled out is required when your code does not compile on windows).